SDCD-1411: Add explicit mention to '**' pathspec to allow whole repository monitoring (#30142)
DataDog/documentation: content/en/continuous_delivery/features/code_changes_detection.md

If the source code patterns for a service are defined in both a link and an extension, only the extension is considered when filtering the commits.

> #### Use service file path patterns to track the whole repository

Reviewer: ```suggestion
#### Use service file path patterns to track changes across the entire repository
```

> If you wish to detect changes in the whole repository, it is possible through the appropiate use of file path patterns. In the example below, `"**"` is used. 

Reviewer: ```suggestion
To detect changes across the entire repository, use appropriate file path patterns. For example, `"**" `matches all files.
```

**Example (schema version v2.2):**

```yaml
extensions:
  datadoghq.com/cd-visibility:
    source_patterns:
      - "**"
```

> In this case, Code Changes Detection for deployments of the `shopist` service will consider all the commits in the repository's Git history.

Reviewer: ```suggestion
In this case, Code Changes Detection for deployments of the `shopist` service includes all commits in the repository's Git history.
```

> <div class="alert alert-warning">If the pattern consists solely of or starts with <code>**</code>, it should be enclosed in quotes, since the * character is reserved for referencing YAML anchors.</div>

Reviewer: ```suggestion
<div class="alert alert-warning">If a pattern is exactly <code>**</code> or begins with it, enclose it in quotes, as <code>*</code> is reserved in YAML for anchors.</div>
```
